src/Sulu/Bundle/AdminBundle/Resources/js/services/Router/Router.js

```diff
@@ -0,0 +1,84 @@
+// @flow
+import {action, autorun, computed, observable} from 'mobx';
+import pathToRegexp, {compile} from 'path-to-regexp';
+import type {Route} from './Route';
+import routeStore from './stores/RouteStore';
+
+export default class Router {
+    history: Object;
+    @observable currentRoute: Route;
+    @observable currentParameters: Object;
+
+    constructor(history: Object) {
+        this.history = history;
+
+        this.history.listen((location) => {
+            this.match(location.pathname);
+        });
+
+        autorun(() => {
+            const path = this.url;
+            if (path !== this.history.location.pathname) {
+                this.history.push(path || this.history.location.pathname);
```

Comment:
This line is causing this error at the beginning:

![image](https://user-images.githubusercontent.com/405874/27382382-008222a8-5687-11e7-9354-785516c6ba31.png)

I somehow need that line, because i also fire the event we listen using `this.history.listen`. Adding a special case for that even adds some overhead, and there already was [some discussion](https://github.com/ReactTraining/react-router/issues/4467) if this is a valid behavior.  The thing is that it's not even that easy to fix that (there are easily some circular references, which will make problems then).